### PR TITLE
Fixed some issues in VtsHalMediaC2V1_0

### PR DESCRIPTION
**Pull request by TianmiChen**
This change includes,
1. Add API to query max and min resolution supported in encoder
2. Pass EOS flags if any in case codec2.0 failed unexpectedly
3. Return correct status for component reset
4. Add sanity check if invalid buffer sent to encoder
5. Refined bitrate control mode
6. Removed some tabs

Tracked-On: OAM-102693
Signed-off-by: Chen, Tianmi <tianmi.chen@intel.com>

### CODE REVIEW

**Comment by TianmiChen** [comment]
@dyang23 , @yuanjunhuang, @zhangyichix , could you please take a review?

Thanks,

**Review by dyang23 — APPROVED** [review]
LGTM, Thanks.
@zhangyichix , please run encode CTS with this patch, make sure no regression before we merge it.
Thanks.

**Comment by TianmiChen** [comment]
> LGTM, Thanks. @zhangyichix , please run encode CTS with this patch, make sure no regression before we merge it. Thanks.

I have run android.media.cts.VideoEncoderTest and android.media.cts.VideoCodecTest to make sure no regression introduced. Yichi can help run the full media cts with latest build. :)

File: c2_components/src/mfx_c2_encoder_component.cpp [hunk 1]
```diff
@@ -278,6 +286,36 @@ void MfxC2EncoderComponent::RegisterClass(MfxC2ComponentsRegistry& registry)
         &MfxC2Component::Factory<MfxC2EncoderComponent, EncoderType>::Create<ENCODER_VP9>);
 }
 
+void MfxC2EncoderComponent::getMaxMinResolutionSupported(
```
Comment (zhangyichix) [inline c2_components/src/mfx_c2_encoder_component.cpp:289]: We can get the supported size from media_codecs_intel_c2_video.xml by MfxXmlParser. Why must have to add this function?

**Comment by dyang23** [comment]
> [ View changes ](/projectceladon/MediaSDK_C2/pull/45/files/77ee3f7668f8434208e95b06cd73bca1b1d94d42)

Great! Since we have already work out a stable plugin, the idea is to merge patch more carefully to avoid regression.

File: c2_components/src/mfx_c2_encoder_component.cpp (hunk 1 above)
Comment (TianmiChen) [reply c2_components/src/mfx_c2_encoder_component.cpp:289]: This is a good concern. Yes, We can get the max and min video size from the config file. But this is for MediaCodec API.

For this vts test, it requested max and min video size via codec2.0 API. If the app cannot query it, the vts will fail. Please check the details in below link.
http://inlubt0232.iind.intel.com:8080/source/xref/CELADON/frameworks/av/media/codec2/hidl/1.0/vts/functional/video/VtsHalMediaC2V1_0TargetVideoEncTest.cpp?r=15ccfa68#461

**Comment by TianmiChen** [comment]
> > [ View changes ](/projectceladon/MediaSDK_C2/pull/45/files/77ee3f7668f8434208e95b06cd73bca1b1d94d42)
> 
> Great! Since we have already work out a stable plugin, the idea is to merge patch more carefully to avoid regression.

Understood! Yes, our current design can make cts pass. But many test cases in vts are failing. The change in this PR is to fix all the failed cases in both VtsHalMediaC2V1_0TargetVideoEncTest and VtsHalMediaC2V1_0TargetVideoDecTest.

**Comment by zhangyichix** [comment]
> 

Ok, thanks for the explanation.

**Review by zhangyichix — APPROVED** [review]

**Pull request merged**